### PR TITLE
Add build guidance for mac users

### LINKED ISSUE #96: Build guidance for mac users
**Description**

The doc now have no guidance for different platforms

**Proposal**

The project docs now don't have any build guidance for different platforms other than Ubuntu. We can add these docs to help the users solving problems when they build on their own dev env.

### PR DESCRIPTION
**Pull request by tianlinzx**
# Which issue does this PR close?

Closes #96

# Rationale for this change
 

# What changes are included in this PR?
I have scripted the prerequirements and tested on MacOS Monterey.

# Are there any user-facing changes?

# How does this change test

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -33,6 +33,17 @@ apt install git curl gcc g++ libssl-dev pkg-config cmake
 
 It should be noted that the compilation of the project has version requirements for dependencies such as cmake, gcc, g++, etc. If your development environment is an old Linux distribution, it is necessary to manually install these dependencies of a higher version.
 
+#### Dependencies(MacOS)
+If the development environment is MacOS, execute the following command to install the required dependencies:
+1.Install XCode from AppStore and install command line tools
```
[the remark this answers is not in the document]
Comment (tianlinzx) [reply README.md]: Submitted another commit according to your comment , would you please review again ?

File: README-CN.md [hunk 2]
```diff
@@ -47,6 +47,18 @@ apt install git curl gcc g++ libssl-dev pkg-config cmake
 
 需要注意的是，项目的编译对 cmake、gcc、g++ 等实际上都是有版本要求的，如果开发环境使用的是较老的 Linux 发行版的话，一般需要手动安装较高版本的这些依赖。
 
+
+#### 开发依赖（MacOS）
+开发环境这里以 MacOS Monterey 为例，执行如下的命令，即可安装好所需的依赖：
+1.安装命令行工具
```
Comment (ShiKaiWi) [inline README-CN.md:53]:
```suggestion

1. 安装命令行工具：
```

File: README-CN.md [hunk 3]
```diff
@@ -47,6 +47,18 @@ apt install git curl gcc g++ libssl-dev pkg-config cmake
...
+```shell
+xcode-select --install
+```
+2. 通过Brew安装cmake
```
Comment (ShiKaiWi) [inline README-CN.md:57]:
```suggestion
2. 通过Brew安装cmake：
```

File: README.md [hunk 4]
```diff
@@ -33,6 +33,17 @@ apt install git curl gcc g++ libssl-dev pkg-config cmake
 
 It should be noted that the compilation of the project has version requirements for dependencies such as cmake, gcc, g++, etc. If your development environment is an old Linux distribution, it is necessary to manually install these dependencies of a higher version.
 
+#### Dependencies(MacOS)
+If the development environment is MacOS, execute the following command to install the required dependencies:
+1.Install command line tools
```
Comment (ShiKaiWi) [inline README.md:38]:
```suggestion

1. Install command line tools:
```

File: README.md [hunk 5]
```diff
@@ -33,6 +33,17 @@ apt install git curl gcc g++ libssl-dev pkg-config cmake
...
+```shell
+xcode-select --install
+```
+2. Install cmake via
```
Comment (ShiKaiWi) [inline README.md:42]:
```suggestion
2. Install cmake:
```

**Review by ShiKaiWi — COMMENTED** [review]
@tianlinzx Thansk for your contributions. I have left some tiny suggestions and would you like to take a look.

File: README.md (hunk 5 above)
Comment (tianlinzx) [reply README.md]: Have made changes and committed again

**Comment by tianlinzx** [comment]
@ShiKaiWi I have acceptted all your changes and committed again.

**Review by ShiKaiWi — APPROVED** [review]
@tianlinzx Thanks!

**Pull request merged**